interfaces/builtin/netlink_driver.go

```diff
@@ -0,0 +1,148 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/interfaces/seccomp"
+	"github.com/snapcore/snapd/snap"
+)
+
+const netlinkDriverSummary = `allows operating a kernel driver module exposing itself via a netlink protocol family`
+
+const netlinkDriverBaseDeclarationSlots = `
+  netlink-driver:
+    allow-installation:
+      slot-snap-type:
+        - core
+        - gadget
+    deny-auto-connection: true
+`
+
+// netlinkDriverInterface type
+type netlinkDriverInterface struct {
+	commonInterface
+}
+
+const netlinkDriverConnectedPlugApparmor = `
+# allow accessing the Linux kernel custom netlink protocol
+# this allows all netlink protocol communication - further 
+# confinement for particular families/protocols is 
+# implemented via seccomp filtering
+network netlink raw,
```

Comment:
`raw` socket usage should not be required IMO - `netlink` should be sufficient so this should be removed unless there is a specific use-case for this already identified.

Also as can be seen in the existing [`netlink-connector` interface](https://github.com/snapcore/snapd/blob/fb315901a3e342c5f13f717e39abff98dee5b74b/interfaces/builtin/netlink_connector.go#L48), `CAP_NET_ADMIN` is required for some particular types of netlink socket usage - to keep this interface consistent with `netlink-connector` I think we should also add

    capability net_admin

here as well.

Reply:
added

Reply:
Thanks